esp_utils.py

```
@@ -2,28 +2,47 @@
         command: The command to run
 
     Returns:
         Tuple[int, str, str]: Return code, stdout, stderr
     """
+    import subprocess
+    
     try:
-        process = await asyncio.create_subprocess_shell(
+        # Use subprocess.run for synchronous execution
+        # Use UTF-8 encoding with error handling for cross-platform compatibility
+        result = subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)